### PR TITLE
Add support for Qt 5.6

### PR DESCRIPTION
**Pull request by net147**

### CODE REVIEW

**Comment by uwerat** [comment]
Most of the code base should stable and - beside of new Rfb specs - I do not expect that the code needs to be changed often. So heaving some #ifdefs for should not generate too much extra work. So in general I'm open for supporting outdated Qt versions.

As a first step I added support for Qt >= 5.10 ( did not verify the early Qt6 versions ) - simply because the changes are kind of trivial. For the others I will comment inline.

What exactly is your motivation for using Qt 5.6 as minimum - the LGPL v2 ?

File: src/VncClient.cpp [hunk 1]
```diff
@@ -119,7 +124,9 @@ namespace
 
 namespace Rfb
 {
+#if QT_VERSION >= QT_VERSION_CHECK(5, 8, 0)
```
Comment (uwerat) [inline src/VncClient.cpp:127]: moc is only for the enum to string conversion used for debugging. For early Qt versions, where moc is not processing namespaces, we have 2 options:

- debug the enums as ints
- make the namespace a dummy class RfbData being derived from QObject

File: src/VncClient.cpp [hunk 2]
```diff
@@ -364,9 +382,15 @@ void VncClient::processClientData()
...
             {
-                auto r = QRandomGenerator::system();
                 m_data->challenge.resize( 16 );
+#if QT_VERSION >= QT_VERSION_CHECK(5, 10, 0)
+                auto r = QRandomGenerator::system();
                 r->generate( m_data->challenge.begin(), m_data->challenge.end() );
+#else
+                typedef std::independent_bits_engine<std::default_random_engine, CHAR_BIT, uchar> random_bytes_engine;
+                random_bytes_engine randomEngine;
+                std::generate( m_data->challenge.begin(), m_data->challenge.end(), std::ref(randomEngine) );
```
Comment (uwerat) [inline src/VncClient.cpp:392]: Isn't std::generate a C++-17 feature, while Qt 5.6 only requires C++-11 ?
Maybe it is better to use qrand() instead ?

**Comment by net147** [comment]
> Most of the code base should stable and - beside of new Rfb specs - I do not expect that it needs to be changed often. So heaving some #ifdefs should not generate too much extra work. So in general I'm open for supporting outdated Qt versions.
> 
> As a first step I added support for Qt >= 5.10 ( did not verify the early Qt6 versions ) - simply because the changes are kind of trivial. For the others I will comment inline.
> 
> What exactly is your motivation for using Qt 5.6 as minimum - the LGPL v2 ?

Yes, Qt 5.6 is last LGPLv2.1 release.

File: src/VncClient.cpp (hunk 2 above)
Comment (net147) [reply src/VncClient.cpp:392]: Yes, you are right.

File: src/VncClient.cpp (hunk 1 above)
Comment (net147) [reply src/VncClient.cpp]: I made a RfbData class.

**Comment by uwerat** [comment]
> Yes, Qt 5.6 is last LGPLv2.1 release.

There is an interesting speech from Burkhard ( IIRC it was this one: https://www.youtube.com/watch?v=bwTlCBbB3RY ), where he explained the details about using LGPL V3 libraries on consumer or business devices. In case your product falls into the business category you might want to migrate to a higher Qt version.

By the way: how did you find this project - I did not make much promotion ?

**Comment by net147** [comment]
> > Yes, Qt 5.6 is last LGPLv2.1 release.
> 
> There is an interesting speech from Burkhard ( IIRC it was this one: https://www.youtube.com/watch?v=bwTlCBbB3RY ), where he explained the details about using LGPL V3 libraries on consumer or business devices. In case your product falls into the business category you might want to migrate to a higher Qt version.
> 
> By the way: how did you find this project - I did not make much promotion ?

Googled: qt vnc server.

**Pull request merged**